src/derivative_operators/vector_calculus_convolutions.jl

```diff
@@ -0,0 +1,343 @@
+# mul! implementation for the case when output array contains vector elements 
+
+function LinearAlgebra.mul!(x_temp::AbstractArray{T1,N2}, has_vector::Bool, A::DerivativeOperator{T2,N}, M::AbstractArray{T3,N2}; overwrite = false) where {T1,T2,T3,N,N2}
+
+    # Check that x_temp has valid dimensions, allowing unnecessary padding in M
+    v = zeros(ndims(x_temp))
+    v .= 2
+    @assert all(([size(x_temp)...] .== [size(M)...])
+        .| (([size(x_temp)...] .+ v) .== [size(M)...])
+        )
+
+    # Check that axis of differentiation is in the dimensions of M and x_temp
+    ndims_M = ndims(M)
+    @assert N <= ndims_M
+    @assert size(x_temp, N) + 2 == size(M, N) # differentiated dimension must be padded
+
+    alldims = [1:ndims(M);]
+    otherdims = setdiff(alldims, N)
+
+    idx = Any[first(ind) for ind in axes(M)]
+    nidx = length(otherdims)
+
+    dims_M = [axes(M)...]
+    dims_x_temp = [axes(x_temp)...]
+    minimal_padding_indices = map(enumerate(dims_x_temp)) do (dim, val)
+        if dim == N || length(dims_x_temp[dim]) == length(dims_M[dim])
+            Colon()
+        else
+            dims_M[dim][begin+1:end-1]
+        end
+    end
+    minimally_padded_M = view(M, minimal_padding_indices...)
+
+    itershape = tuple(dims_x_temp[otherdims]...)
+    indices = Iterators.drop(CartesianIndices(itershape), 0)
+
+    setindex!(idx, :, N)
+    for I in indices
+        # replace all elements of idx with corresponding elt of I, except at index N
+        Base.replace_tuples!(nidx, idx, idx, otherdims, I)
+        mul!(view(x_temp, idx...), true, A, view(minimally_padded_M, idx...), overwrite = false)
+    end
+end
+
+##################################################################################
+# Divergence and Gradient convolutions
+##################################################################################
+
+function LinearAlgebra.mul!(x_temp::AbstractVector, has_vector::Bool, A::DerivativeOperator,x::AbstractVector; overwrite = false)
+    convolve_BC_left!(x_temp, A, x, overwrite = overwrite)
+    convolve_interior!(x_temp, A, x, overwrite = overwrite)
+    convolve_BC_right!(x_temp, A, x, overwrite = overwrite)
+end
+
+function convolve_interior!(x_temp::AbstractVector{T1},  A::DerivativeOperator{T3,N,false}, x::AbstractVector{T2}; overwrite = true) where {T1, T2, T3, N}
```

Comment:
This dispatch already exists?

Reply:
I needed a way to divert the convolutions for vcops from those defined for derivate operators since both of them accept similar arguments but behave really differently. Thats why this dispatch (the function at the start), it has an additional dummy argument and will lead to the convolutions specifically meant for vcops.